packit/source_git.py

```diff
@@ -514,18 +512,33 @@ def _rebase_patches(self, from_branch):
         BUILD_dir = self.get_BUILD_dir()
         self.local_project.fetch(BUILD_dir, f"+{from_branch}:{to_branch}")
 
-        # shorter format for better readability in case of an error
-        commits_to_cherry_pick = [
-            c.hexsha[:8]
-            for c in LocalProject(working_dir=BUILD_dir).get_commits(from_branch)
-        ][-2::-1]
-        if commits_to_cherry_pick:
+        # transform into patch_name -> index
+        patches = {
+            Path(p.path).name: p.index
+            for p in self.dist_git.specfile.patches.get("applied", [])
+        }
+
+        # -2 - drop first commit which represents tarball unpacking
+        # -1 - reverse order, HEAD is last in the sequence
+        patch_commits = list(
+            LocalProject(working_dir=BUILD_dir).get_commits(from_branch)
+        )[-2::-1]
+
+        for commit in patch_commits:
             self.local_project.git_repo.git.cherry_pick(
-                *commits_to_cherry_pick,
+                commit.hexsha,
                 keep_redundant_commits=True,
                 allow_empty=True,
                 strategy_option="theirs",
             )
+
+            # Annotate commits in the source-git repo with patch_id. This info is not provided
+            # during the rpm patching process so we need to do it here.
+            metadata = PatchMetadata.from_commit(commit=commit)
+            # commit.message already ends with \n
+            message = f"{commit.message}patch_id: {patches[metadata.name]}"
```

Comment:
`metadata.name`? This doesn't seem to be correct.

Reply:
why not? it's name of the patch

Reply:
Okay, sorry, it was not obvious from the name that `patches` is a patch name - patch index mapping. Maybe rename it to `patch_ids` or something similar?

Reply:
`patch_name_ids` seems confusing, as it can be read as the "the ids of the patch names", but it's really the "the ids of the patches". I still think `patch_ids` would be the most expressive here.

I'm spending so much time on this, b/c bad naming distracts the reader, and can make understanding the code needlessly difficult.